# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.cpp
# Commit: acafbde0a546
# Full Hash: acafbde0a54602fcd1a634ddd772d666f7e96455
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-03-09 16:11:38
# Regressor Bug: 1696266
# File Overlap Count: 2
# Description:
#   Bug 1696266, limit the load types which may cause the page to enter bfcache, r=peterv
#   
#   The change to dom/base/nsFrameLoaderOwner.cpp is to log about the issues but still ensure we don't crash.
#   
#   I'd prefer to not put error loads to bfcache.
# ==============================================================================

diff -r f0d286830ebb -r acafbde0a546 docshell/base/nsDocShell.cpp
--- a/docshell/base/nsDocShell.cpp	Tue Mar 09 13:34:51 2021 +0000
+++ b/docshell/base/nsDocShell.cpp	Tue Mar 09 13:42:43 2021 +0000
@@ -6951,7 +6951,6 @@
   // Only save presentation for "normal" loads and link loads.  Anything else
   // probably wants to refetch the page, so caching the old presentation
   // would be incorrect.
-  // XXXBFCache in parent needs something like this!
   if (aLoadType != LOAD_NORMAL && aLoadType != LOAD_HISTORY &&
       aLoadType != LOAD_LINK && aLoadType != LOAD_STOP_CONTENT &&
       aLoadType != LOAD_STOP_CONTENT_AND_REPLACE &&